src/FirebaseArduino.h

```diff
@@ -24,18 +24,97 @@
 #define FIREBASE_JSONBUFFER_SIZE 200
 #endif // FIREBASE_JSONBUFFER_SIZE
 
+/**
+ * Gateway class for Arduino clients to interact with the Firebase backend.
+ * This implementation is designed to follow Arduino best practices and favor
+ * simplicity over all else. 
+ * For more complicated usecases and more control see the Firebase class in 
+ * Firebase.h.
+ */
 class FirebaseArduino {
  public:
+  /**
+   * Must be called first. This sets the class up for use.
+   * \param host Your firebase db domain name, usually X.firebaseIO.com.
+   * \param auth Optional authentication for the db, a secret or token.
+   */
   void begin(const char* host, const char* auth = "");
+
+  /**
+   * Adds the data in value to the list located at path. Equivilent to the
```

Comment:
Updated it I find the official docs description confusing. I tried to blend them though. 

I don't see an inline way to specify \code it looks like you can add whole code blocks like:
/**
* \code
*   String push(...)
* \endcode 
*/
But I don't think you can do part of the line. Maybe we can play with it once we can see the results.